app/src/main/java/ch/epfl/skysync/viewmodel/InFlightViewModel.kt

```diff
@@ -372,7 +378,18 @@ class InFlightViewModel(val repository: Repository) : ViewModel() {
 
   /** Save the finished flight to the database */
   private suspend fun saveFinishedFlight() {
-    // TODO: save the finished flight to the database
+    if (currentFlight.value == null) {
+      return
+    }
+    val finishedFlight =
+        currentFlight.value!!.finishFlight(
+            takeOffTime = takeOffTime!!,
+            takeOffLocation = _flightLocations.value.first().point,
+            landingTime = landingTime!!,
+            landingLocation = _flightLocations.value.last().point,
+            flightTime = ChronoUnit.MILLIS.between(takeOffTime, landingTime),
+            flightTrace = FlightTrace(trace = _flightLocations.value.map { it.point }))

```

Comment:
While the `flightTrace` is non nullable and thus you should pass a value, it won't be used to create the flight trace in the database, that is already done in `saveFlightTrace`, so we could keep it as it is, but this could be misleading, or we could pass an empty instance of FlightTrace and add a comment that state why we need to pass an instance here